deepchem/data/datasets.py

```diff
@@ -2500,28 +2687,34 @@ def transform(self, transformer: "dc.trans.Transformer",
 
     Parameters
     ----------
-    transformer: Transformer
-      the transformation to apply to each sample in the dataset
+    transformer: dc.trans.Transformer
+      The transformation to apply to each sample in the dataset
 
     Returns
     -------
-    a newly constructed Dataset object
+    ImageDataset
+      A newly constructed ImageDataset object
     """
     newx, newy, neww, newids = transformer.transform_array(
         self.X, self.y, self.w, self.ids)
-    return NumpyDataset(newx, newy, neww, newids)
+    return ImageDataset(newx, newy, neww, newids)
```

Comment:
This is incorrect.  `transform_array()` returns numpy arrays.

Reply:
OK. I revert this modifications